### ISSUE TITLE
Global attention in key_padding_mask

### ISSUE THREAD

**Issue by greenstars** [allenai/longformer#147]
Hi, 

I'm trying to understand the attention_mask=key_padding_mask.unsqueeze(dim=1).unsqueeze(dim=1) * -1 in class LongformerSelfAttentionForBart. 'key_padding_mask' has only True or False as far as I understand due to 
"def invert_mask(attention_mask):
    """Turns 1->0, 0->1, False->True, True-> False"""
    assert attention_mask.dim() == 2
    return attention_mask.eq(0)"
in modeling_bart.py.  It would be great if you could point me to the part of the code how it adds global attention while we go through modeling_bart for Longformer+Bart. 

Thanks,
Akila

**Comment by elb3k** [comment]
In [test_readme.py](https://github.com/allenai/longformer/blob/55c1f5f75d6f33b7ee23af0caf0392a52f8d4f4c/tests/test_readme.py#L51) there is an example with using local, global and no attention.

But it is a little misleading, test_readme.py states that:
`# Attention mask values -- 0: no attention, 1: local attention, 2: global attention`

And in [longformer.py](https://github.com/allenai/longformer/blob/55c1f5f75d6f33b7ee23af0caf0392a52f8d4f4c/longformer/longformer.py#L98):
```
'''
The `attention_mask` is changed in `BertModel.forward` from 0, 1, 2 to
    -ve: no attention
      0: local attention
    +ve: global attention
'''
```

I found the first definition works.  So mask is torch.Tensor with dimensions `batch, seq_len, 1` with `0 - no attention, 1 - local attention, 2 - global attention`

**Comment by tangxxiii** [comment]
I have the same question. Specifically, after being processed by invert_mask() function, the attention mask becomes [0,0,..,1,1], where 0s are local attentions and global attentions, and 1s are no attention. I wonder how does this attention mask work on LongformerSelfAttention.
Have you figured it out?

**Comment by Trace2333** [comment]
Same problem, I searched all code in longformer and the related code in transformers,the invert_mask() function in BartEncoder destroies the integer 2 in the attention mask,but their code regard the mask as it dosen't  been converted.
